pkg/api/live/conn.go

```diff
@@ -106,7 +112,9 @@ func (c *connection) writePump() {
 		select {
 		case message, ok := <-c.send:
 			if !ok {
-				c.write(websocket.CloseMessage, []byte{})
+				if err := c.write(websocket.CloseMessage, []byte{}); err != nil {
+					log.Warn("Failed to write close message to connection: %s", err)
```

Comment:
Need to be `Warnf` to work. But use argument `log.Warn("Failed to write close message to connection", "err", err` instead.

Reply:
To me it looks as if `log.Warn` in this case is actually taking a format string: `func Warn(format string, v ...interface{})`. It's confusing to me as well, but apparently different loggers throughout the code have different APIs. I'd prefer having a consistent argument pair API.

Reply:
For this specific case I would add a `log.Logger` to the `connection` struct. Easiest maybe to require a `log.Logger` instance in the `newConnection` factory method?

Reply:
@marefr OK I'll try.

Reply:
@marefr I pushed a change introducing a field `conn::log`, where `log` gets instantiated within `newConnection`. That look OK to you?

I'm also wondering if I should change all of conn.go to use the `log` field.